---
title: Software

# The About page
# v2.0
# https://github.com/cotes2020/jekyll-theme-chirpy
# Â© 2017-2019 Cotes Chung
# MIT License
---
<head>
  <link
    href="https://fonts.googleapis.com/css?family=Montserrat"
    rel="stylesheet"
  />
  <link rel="stylesheet" href="../../assets/css/main.css" />
</head>

# Software

Here is the list of open-source documented software hosted on GitHub that my lab developed or co-developed. Some of these are being actively maintained and used by 1000+ users.

- [Winnowmap](https://github.com/marbl/Winnowmap) - Long read (ONT/PacBio) mapper using weighted minimizer sampling technique <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/marbl/Winnowmap)

- [Mashmap](https://github.com/marbl/MashMap) - fast approximate mapper for long DNA sequences (e.g., reads, contigs) using LSH <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/marbl/MashMap)

- [FastANI](https://github.com/ParBLiSS/FastANI) - fast whole-genome similarity (ANI) estimator <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/ParBLiSS/FastANI)

- [VF](https://github.com/at-cg/VF) - Provably-good algorithms for variant filtering in genome graphs <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/at-cg/VF)

- [ChainX](https://github.com/AT-CG/ChainX) - Optimal co-linear anchor chaining algorithm <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/AT-CG/ChainX)

- [Metamaps](https://github.com/DiltheyLab/MetaMaps) - strain-level metagenomics analysis for long reads <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/DiltheyLab/MetaMaps)

- [PaSGAL](https://github.com/ParBLiSS/PaSGAL) - parallel (multicore-SIMD) long/short reads to pan-genome graph (DAG) alignment <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/ParBLiSS/PaSGAL)

- [ParConnect](https://github.com/ParBLiSS/parconnect) - distributed-memory parallel connected components solver for massive graphs <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/ParBLiSS/parconnect)

- [PairG](https://github.com/ParBLiSS/PairG) - index builder for validating paired-end alignment distance constraints in pan-genome graphs <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/ParBLiSS/PairG)

- [Kmerind](https://github.com/ParBLiSS/kmerind) - Parallel k-mer indexing for biological sequences <br>
[![Code](https://img.shields.io/badge/Source-Code-blue)](https://github.com/ParBLiSS/kmerind)
